hbase-rest/src/main/java/org/apache/hadoop/hbase/rest/ScannerResource.java

```diff
@@ -59,8 +62,25 @@ public ScannerResource(TableResource tableResource) throws IOException {
     this.tableResource = tableResource;
   }
 
+  private static Cache<String, ScannerInstanceResource> setupScanners() {
+    final Configuration conf = HBaseConfiguration.create();
+
+    int size = conf.getInt(REST_SCANNERCACHE_SIZE, DEFAULT_REST_SCANNERCACHE_SIZE);
+    long evictTimeoutMs = conf.getTimeDuration(REST_SCANNERCACHE_EXPIRE_TIME,
+      DEFAULT_REST_SCANNERCACHE_EXPIRE_TIME_MS, TimeUnit.MILLISECONDS);
+
+    Cache<String, ScannerInstanceResource> cache =
+        Caffeine.newBuilder()
```

Comment:
Any reason we are not using org.apache.hbase.thirdparty.com.google.common.cache.CacheBuilder?

Reply:
The Guava docs itself suggest using this one instead.

Reply:
This one is supposed to be better (though perf doesn't matter that much in this case), and the the docs imply that the Guava one is not actively developed anymore.
We already use this library elsewhere.